Sources/Datadog/Core/Upload/DataUploader.swift

```diff
@@ -6,32 +6,52 @@
 
 import Foundation
 
-/// URL of the endpoint for data uploads. It's being validated during its creation.
-internal struct DataUploadURL {
-    let url: URL
+/// Creates URL and adds query items before providing them
+internal class UploadURLProvider {
+    private let rawURL: URL
+    private let dateProvider: DateProvider
 
-    init(endpointURL: String, clientToken: String) throws {
+    private var queryItems: [URLQueryItem] {
+        // batch_time
+        let currentDateInMs = dateProvider.currentDate().timeIntervalSinceReferenceDate
```

Comment:
1/ Should use `. timeIntervalSinceReferenceDate()`? AFAIK it's the interval since 1st January of 2000. Shouldn't it be timestamp since 1970? 
2/ Also, this is in seconds, does backend require seconds or milliseconds / microseconds?

Reply:
i verified with https://github.com/DataDog/dd-sdk-android/pull/181/files
they are using [`currentTimeMillis` from Java](https://docs.oracle.com/javase/7/docs/api/java/lang/System.html#currentTimeMillis())

https://github.com/DataDog/dd-sdk-android/blob/master/dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/time/DatadogTimeProvider.kt

changed ✅